lib/OriginalSource.js

```diff
@@ -9,6 +9,9 @@ const splitIntoLines = require("./helpers/splitIntoLines");
 const getGeneratedSourceInfo = require("./helpers/getGeneratedSourceInfo");
 const Source = require("./Source");
 const splitIntoPotentialTokens = require("./helpers/splitIntoPotentialTokens");
+const {
+	isDualStringBufferCachingEnabled
+} = require("./helpers/stringBufferUtils");
 
 class OriginalSource extends Source {
 	constructor(value, name) {
```

Comment:
I think we want to internal OriginalSource `value` too?

Reply:
I intentionally limited string interning just to `RawSource` for now since during my experiments, it seemed like memory may increase if applying the interning in other source objects (if it retains memory beyond the lifetime of the source object). I would need to go back and more rigorously determine the behavior for each class to understand where the memory is coming from, but at least for `RawSource` this was an improvement.